frontend/src/metabase/css/dashboard.css

```diff
@@ -162,6 +162,27 @@
   box-shadow: 3px 3px 8px var(--color-shadow);
 }
 
+.TextCard .has-no-content {
```

Comment:
Is it possible to move these styles to Text/Heading components? Seems like `.TextCard` class is assigned in the `DashboardGrid` in `renderGridItem` which sort of broadens its responsibility to not only being just the layout of abstract cards but also knowing what cards are and which classes we should assign

Reply:
The reason these are not in Text/Heading components themselves is that `react-grid-layout` applies styles to its components via CSS classes. Certain styles are only customizable by targeting the CSS class applied to the them by `react-grid-layout`.

So, in order to change the color of the resizing handle
![image](https://github.com/metabase/metabase/assets/22608765/f5046825-c226-4b94-8601-e31e55c38a0c)
and styling applied during "resizing"
![resizing-styling](https://github.com/metabase/metabase/assets/22608765/1d30a66e-f36a-4497-96f2-2375a23311b4)
we need to be able to target `.react-resizable-handle:after` and `.resizing`. The `.TextCard` class is chained before them to further specify that we want to apply these styles only to text/heading cards - not other types of dash-cards. 

Also I used
```css
.TextCard:hover .text-edit-container,
.TextCard:focus-within .text-edit-container {
  border-radius: 8px;
  border: 1px solid var(--color-brand);
}
```
because `react-grid-layout` was consuming/stopping the propagation of hover events over its draggable surface area such that only hovering over the inner-portion of a card applied correct border styling, but we want the whole card to display a border when hovered.

Of course, this below could be moved to the text/heading components themselves b/c it is not affected by `react-grid-layout`, but then we would have this kind of styling in multiple places which seemed like a bad/confusing idea to me.
```css
.TextCard .has-no-content {
  border-radius: 8px;
  border: 1px solid var(--color-brand);
}
```

Reply:
We could eliminate `text-edit-container` global class and move the logic to the `Heading` component. If it needs to be reusable between Text and Heading cards, we could extract a `TextCardContainer`. But yes, to style the drag handle we need some global css

ts```
import { css } from "@emotion/react";
import styled from "@emotion/styled";
import { color } from "metabase/lib/colors";

interface InputContainerProps {
  isPreviewing: boolean;
  isEmpty: boolean;
}

export const InputContainer = styled.div<InputContainerProps>`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  justify-content: center;
  overflow: hidden;
  padding: 0 0.65em;
  padding-left: 0;
  pointer-events: auto;
  border-radius: 8px;

  .DashCard:hover &,
  .DashCard:focus-within & {
    border: 1px solid ${color("brand")};
  }

  .DashCard.react-draggable-dragging &,
  .DashCard.react-resizable-resizing & {
    border: unset;
  }

  ${({ isPreviewing }) =>
    !isPreviewing &&
    css`
      padding: 0 0.75em;
    `}

  ${({ isEmpty }) =>
    isEmpty &&
    css`
      border: 1px solid ${color("brand")};
    `}
`;
```